.github/workflows/build-and-test.yml

```diff
@@ -0,0 +1,52 @@
+name: facebook/metro/build-and-test
+on:
+  workflow_call:
+  pull_request:
+    types: [opened, synchronize]
+
+# head_ref is per PR, so this ensures that updaing the latest PR commit
+# will cancel the previous run of the workflow and trigger a new one
+concurrency:
+  group: "build-and-test-${{ github.head_ref }}"
+  cancel-in-progress: true
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  run-js-checks:
+    runs-on: ubuntu-latest
+    name: "Typescript, Link, Smoke Test"
+    steps:
+      - uses: actions/checkout@v4
+      - uses: ./.github/actions/yarn-install
+      - run: yarn typecheck
+      - run: yarn typecheck-ts
+      - run: yarn lint
+      - run: yarn test-smoke
+
+  test-with-coverage:
+    runs-on: ubuntu-latest
+    name: "Test Coverage Calculation"
+    steps:
+    - uses: actions/checkout@v4
+    - uses: ./.github/actions/yarn-install
+    - run: yarn test-coverage
+    - run: "./.github/scripts/install_codecov.sh"
+    - run: "./codecov -t ${{ secrets.CODECOV_TOKEN }} -f ./coverage/coverage-final.json"
+
+  test:
+    strategy:
+      matrix:
+        runs-on: ['ubuntu-latest'] # 'windows-latest'
```

Comment:
IIRC we have a bunch of tests currently not passing on Windows unfortunately and this probably can't be enabled on `main` in the short term, but in CircleCI we have the convention of [`windows` branches to trigger CI on Windows](https://github.com/facebook/metro/blob/e3aaa0bac2d91598646341dfd5621c28cdf7370f/.circleci/config.yml#L133), which is useful if we're looking at a specific test suite.

Alternatively... if we could always run on Windows but have Jest automatically skip known-broken Windows tests *only* on Windows - maybe with a [`--filter`](https://jestjs.io/docs/cli#--filterfile) file, that'd be a much better setup.

Reply:
I was planning to fix the windows tests on a next commit and will enable it. WDYT?

Reply:
Afaik fixing tests on Windows isn't trivial but yeah we can follow up. I'd probably remove the commented-out `'windows-latest'` for now though.